src/main/scala/com/jcdecaux/setl/transformation/FactoryInput.scala

```diff
@@ -17,4 +17,7 @@ import scala.reflect.runtime
  */
 private[setl] case class FactoryInput(override val runtimeType: runtime.universe.Type,
                                       producer: Class[_],
-                                      deliveryId: String = Deliverable.DEFAULT_ID) extends HasType
+                                      deliveryId: String = Deliverable.DEFAULT_ID,
+                                      autoLoad: Boolean,
+                                      optional: Boolean,
+                                      consumer: Class[_ <: Factory[_]]) extends HasType
```

Comment:
Following the comment in `FactoryDeliveryMetadata`, maybe it's more understandable if we rename the filed `consumer` to `factoryClass`